### PR TITLE
[CLD-4578] Add new module provisioner read-only user

### PR DESCRIPTION
**Pull request by andrleite**
Issue: CLD-4578
Signed-off-by: Andre Leite [andrleite@gmail.com](mailto:andrleite@gmail.com)

#### Summary

This PR adds a new module for read-only user creation. I've kept permissions granular.

#### Ticket Link
https://mattermost.atlassian.net/browse/CLD-4578

#### Release Note

```release-note
NONE
```

### CODE REVIEW

**Review by Mshahidtaj — DISMISSED** [review]
Good Work @andrleite . minor comment , would be good to add usage example as well similar to this https://github.com/mattermost/mattermost-cloud-monitoring/tree/master/aws/community-infra/example

**Comment by andrleite** [comment]
> Good Work @andrleite . minor comment , would be good to add usage example as well similar to this https://github.com/mattermost/mattermost-cloud-monitoring/tree/master/aws/community-infra/example

Thank you for the suggestion, Added.

File: aws/provisioner-read-only-users/variables.tf [hunk 1]
```diff
@@ -0,0 +1,14 @@
+variable "environment" {
+  type        = string
+  description = "The name of the environment is created"
+}
+
+variable "deployment_name" {
```
Comment (spirosoik) [inline aws/provisioner-read-only-users/variables.tf:6]: we are trying to follow a pattern overall if you check other modules.

**Review by spirosoik — APPROVED** [review]
good job

**Review by d-wierdsma — APPROVED** [review]

**Pull request merged**